dot/digest/digest.go

```diff
@@ -259,20 +262,19 @@ func (h *Handler) setBABEDigestsOnFinalization(finalizedHeader *types.Header) er
 	nextEpoch := currEpoch + 1
 
 	err = h.epochState.FinalizeBABENextEpochData(nextEpoch)
-	if err != nil {
-		h.logger.Debugf("failed to store BABENextEpochData for block number %d (%s): %w",
+	if !errors.Is(err, state.ErrEpochNotInMemory) {
+		return fmt.Errorf("cannot finalize babe next epoch data for block number %d (%s): %w",
 			finalizedHeader.Number, finalizedHeader.Hash(), err)
-		return err
 	}
```

Comment:
Potato question, if we get `state.ErrEpochNotInMemory`, should we just continue execution?

Reply:
Yes, we should. First we try to finalize the `next-epoch-data` if the `state.ErrEpochNotInMemory` error occur then we should try to finalize `next-config-data` and if the `state.ErrEpochNotInMemory` error occur again then we return `ErrCannotDefineNextEpoch`. We should finalize at least one next epoch information, normally those info comes in the same block but sometimes only one of them appears